src/testmenu.cpp

```diff
@@ -0,0 +1,778 @@
+            return F("Action:ToggleTRK");
+
+        case MENU_SECONDARY_RATE_1:
+            return F("Action:SetSecDist|0.1");
+        case MENU_SECONDARY_RATE_2:
+            return F("Action:SetSecDist|0.5");
+        case MENU_SECONDARY_RATE_3:
+            return F("Action:SetSecDist|2");
+        case MENU_SECONDARY_RATE_4:
+            return F("Action:SetSecDist|5");
+        case MENU_SECONDARY_RATE_5:
+            return F("Action:SetSecDist|15");
+
+        case MENU_SECONDARY_ALT_UP:
+            return F("Action:MoveALTAxis|UP");
+        case MENU_SECONDARY_ALT_DOWN:
+            return F("Action:MoveALTAxis|DOWN");
+        case MENU_SECONDARY_AZ_LEFT:
+            return F("Action:MoveAZAxis|LEFT");
+        case MENU_SECONDARY_AZ_RIGHT:
+            return F("Action:MoveAZAxis|RIGHT");
+        case MENU_FACTORY_RESET:
+            return F("Action:FactoryReset");
+        case MENU_PASSTHROUGH_COMMAND:
+            return F("Action:PassthroughCmd");
+        case MENU_MAIN_LIST_HARDWARE:
+            return F("Action:ListHardware");
+        case MENU_MAIN_CONNECT_DRIVERS:
+            return F("Submenu:ConnectDrivers");
+        case MENU_MAIN_PRIMARY_AXIS_MOVES:
+            return F("Submenu:PrimaryAxisMoves");
+        case MENU_MAIN_SECONDARY_AXIS_MOVES:
+            return F("Submenu:SecondaryAxisMoves");
+        default:
+            return F("Unknown");
+    }
+}
+TestMenuItem::TestMenuItem(menuText_t labelId, TestMenu *subMenu)
+{
+    _key       = -1;
+    _label     = getMenuLabel(labelId);
+    _action    = getMenuAction(labelId);
+    _isSubMenu = subMenu != nullptr;
+    _subMenu   = subMenu;
+}
+
+void TestMenuItem::display() const
+{
+    Serial.print("  [");
+    Serial.print(_key);
+    Serial.print("] ");
+    Serial.println(_label);
+}
+
+int TestMenuItem::getKey() const
+{
+    return _key;
+}
+
+void TestMenuItem::setKey(int key)
+{
+    _key = key;
+}
+
+String TestMenuItem::getAction() const
+{
+    return _action;
+}
+
+TestMenu *TestMenuItem::getSubMenu() const
+{
+    return _subMenu;
+}
+
+void TestMenu::setParentMenu(TestMenu *parentMenu)
+{
+    _parentMenu = parentMenu;
+}
+
+TestMenu::TestMenu(int level, String name, String parent, TestMenuItem *choices, int numChoices, TestMenu *parentMenu)
+{
+    _lastTick          = 0;
+    _level             = level;
+    _name              = name;
+    _parent            = parent;
+    _choices           = choices;
+    _numChoices        = numChoices;
+    _parentMenu        = parentMenu;
+    _secondaryDistance = 1;
+    if (_currentMenu == nullptr)
+    {
+        _backItem = new TestMenuItem(MENU_BACK);
+        TestMenu::_backItem->setKey(0);
+    }
+    _currentMenu = this;  // Set the last created menu as the current menu
+    for (int i = 0; i < _numChoices; i++)
+    {
+        if (_choices[i].getSubMenu() != nullptr)
+        {
+            _choices[i].getSubMenu()->setParentMenu(this);
+        }
+    }
+}
+String getComponent(String comp)
```

Comment:
const reference to avoid a copy
```suggestion
String getComponent(const String& comp)
```